Makefile

```
@@ -359,17 +359,32 @@ rust-bench: ## Run Rust benchmarks
+    -o addopts="" \
+    -o python_files="benchmark_*.py" \
+    -o python_classes="Benchmark*" \
+    --benchmark-only --benchmark-disable-gc
+
+benchmark: setup-logs ## Run serializer benchmarks + save a local baseline
+	@echo "$(BLUE)Running serializer benchmarks (saving baseline)...$(RESET)"
+	@echo "$(YELLOW)Logging to $(LOG_BENCHMARK_DIR)/bench_$(TIMESTAMP).log$(RESET)"
+	@$(BENCHMARK_PYTEST) --benchmark-autosave 2>&1 | tee $(LOG_BENCHMARK_DIR)/bench_$(TIMESTAMP).log
+	@echo "$(GREEN)✓ Saved to .benchmarks/ (gitignored, per-machine)$(RESET)"
+
+benchmark-compare: setup-logs ## Run benchmarks, fail on >10% median regression vs last saved baseline
```

checkmake (0.3.2)

[warning] Target body for "benchmark-compare" exceeds allowed length of 5 lines (7).

(maxbodylength)